src/main/resources/db/migration/0200/V231__ReportProjects.sql

```diff
@@ -0,0 +1,11 @@
+ALTER TABLE reports DROP CONSTRAINT one_report_per_quarter;
+
+ALTER TABLE reports ADD COLUMN project_id BIGINT REFERENCES projects (id) ON DELETE SET NULL;
+ALTER TABLE reports ADD COLUMN project_name TEXT;
+
+CREATE INDEX ON reports (project_id);
+
+ALTER TABLE reports ADD CONSTRAINT one_project_report_per_quarter
+    UNIQUE (organization_id, project_id, year, quarter);
+ALTER TABLE reports ADD CONSTRAINT one_org_report_per_quarter
+    EXCLUDE (organization_id WITH =, year WITH =, quarter WITH =) WHERE (project_id IS NULL);
```

Comment:
Does this essentially allow dupes only when project id is null (because of line 3)?

Reply:
Unique constraints count nulls as distinct from one another, so the first constraint would, by itself, allow dupes when the project ID is null.

The second constraint says that when the project ID is null, require that the org+year+quarter be unique (in other words, don't allow duplicate org-level reports).

Incidentally, the latest PostgreSQL releases have a new feature that makes this dance unnecessary; you can tell it that nulls shouldn't be treated as distinct from other nulls. But we're using an older PostgreSQL release in production that doesn't have that feature, so we have to split it into two constraints.